engine/src/main/java/org/hibernate/search/query/engine/impl/AbstractHSQuery.java

```diff
@@ -233,6 +242,12 @@ public ExtendedSearchIntegrator getExtendedSearchIntegrator() {
 		return extendedIntegrator;
 	}
 
+	/**
+	 * Returns the names of the projection constants (beginning with {@link #HSEARCH_PROJECTION_FIELD_PREFIX}) supported
+	 * by a specific implementation.
+	 */
+	protected abstract Set<String> getSupportedProjectionConstants();
```

Comment:
And we also have an issue with spatial distance. It starts with only one underscore.